main/bamboo/bamboo.go

```diff
@@ -171,6 +180,67 @@ func listenToZookeeper(conf configuration.Configuration, eventBus *event_bus.Eve
 	return serviceConn
 }
 
+func listenToEventStream(conf configuration.Configuration, eventBus *event_bus.EventBus) {
+	client := &http.Client{}
+	client.Timeout = 0 * time.Second
+
+	for _, marathon := range conf.Marathon.Endpoints() {
+		go func() {
+			for {
+				req, err := http.NewRequest("GET", marathon+"/v2/events", nil)
+				req.Header.Set("Accept", "text/event-stream")
+				if err != nil {
+					errorMsg := "An error occurred while creating request to Marathon events system: %s\n"
+					log.Printf(errorMsg, err)
+					continue
+				}
+
+				resp, err := client.Do(req)
+				if err != nil {
+					continue
+				}
+
+				reader := bufio.NewReader(resp.Body)
+				for {
+					line, err := reader.ReadString('\n')
+					if err != nil {
+						if err != io.EOF {
+							errorMsg := "An error occurred while reading Marathon event: %s\n"
+							log.Printf(errorMsg, err)
+						}
+						break
+					}
+
+					if len(strings.TrimSpace(line)) == 0 {
+						continue
+					}
+
+					if !strings.HasPrefix(line, "data: ") {
+						errorMsg := "Wrong event format: %s\n"
+						log.Printf(errorMsg, line)
+						continue
+					}
+
+					line = line[6:]
+
+					var event event_bus.MarathonEvent
+					err = json.Unmarshal([]byte(line), &event)
+
+					if err != nil {
+						errorMsg := "Unable to decode JSON Marathon Event request: %s\n"
+						log.Printf(errorMsg, line)
+						continue
+					}
+
+					eventBus.Publish(event)
+				}
+
+				_ = resp.Body.Close()
```

Comment:
Instead of closing it here, use `defer resp.Body.Close()` straight after you open it. That'll work best if you split the body of this loop out into a seperate function.

Also, I'd add some logging here; a streaming connection dropping is worth mentioning.